consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -311,6 +315,25 @@ impl Drop for ByzantineLedger {
     }
 }
 
+fn make_metadata_provider(
+    quorum_set: QuorumSet,
+    enclave: impl ReportableEnclave + Send + Sync + 'static,
+    msg_signing_key: Arc<Ed25519Pair>,
+) -> BlockMetadataProvider {
+    Arc::new(move |block_data| {
+        let verification_report = enclave.get_ias_report().expect("failed to get AVR");
+        let contents = BlockMetadataContents::new(
+            block_data.block().id.clone(),
+            quorum_set.clone(),
```

Comment:
This somewhat implies that new nodes join the network start in "catchup mode" and create their own metadata signatures with their own quorum set even for blocks they weren't participating in forming.

I wonder if there's an issue with that i.e. where nodes that didn't participate potentially represent that they did? 

Is there potentially a need to create a hard minimum for when nodes can sign on-chain blocks?

Reply:
This is the requirement I got from James. It turns out that with SCP, each consensus node has its own blockchain, and we want the block signatures and metadata to be from that node. For nodes that are catching up, they supply their own metadata but no signature.